### PR TITLE
Add a simple health checker

### PR DESCRIPTION
**Pull request by tjdevries**
Shougo,

I mentioned in gitter yesterday adding your own deoplete health checker. This PR just adds a very simple one for you. Feel free to just delete everything I wrote and use it as a template if you'd like, but I thought you might like a starting point.

When someone uses `:CheckHealth` with this in your plugin, the output would now look something like:

--------------------------------------------------------------------------------
--------------------------------------------------------------------------------

# Checking health
--------------------------------------------------------------------------------

## Checker health#deoplete#check says:

  - Checking: Deoplete.nvim
    - SUCCESS: has("nvim") was successful
    - SUCCESS: has("python3") was successful
    - INFO: If you're still having problems, try the following commands:

        $ export NVIM_PYTHON_LOG_FILE=/tmp/log

        $ export NVIM_PYTHON_LOG_LEVEL=DEBUG

        $ nvim

        $ cat /tmp/log_{PID}

         and then create an issue on github

--------------------------------------------------------------------------------

## Checker health#nvim#check says:

. . .

--------------------------------------------------------------------------------
--------------------------------------------------------------------------------

If you have any questions, let me know! I'd be glad to help.

Thanks for making such an awesome plugin!

### CODE REVIEW

File: autoload/health/deoplete.vim [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: Please add the header like other files.

File: autoload/health/deoplete.vim [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    call health#report_error('has("python3") was not successful', [
+          \ 'Please install the python3 package for neovim.',
+          \ 'A good guide can be found here: https://github.com/tweekmonster/nvim-python-doctor/wiki/Advanced:-Using-pyenv'
+          \ ]
+          \ )
+  endif
+endfunction
+
+function! s:still_have_issues() abort
+  let l:indentation = '        '
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: It should be `indentation` without `l:`.

File: autoload/health/deoplete.vim [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  endif
+endfunction
+
+function! s:check_required_python_for_deoplete() abort
+  if has('python3')
+    call health#report_ok('has("python3") was successful')
+  else
+    call health#report_error('has("python3") was not successful', [
+          \ 'Please install the python3 package for neovim.',
+          \ 'A good guide can be found here: https://github.com/tweekmonster/nvim-python-doctor/wiki/Advanced:-Using-pyenv'
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: too long line.

File: autoload/health/deoplete.vim [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          \ ]
+          \ )
+  endif
+endfunction
+
+function! s:still_have_issues() abort
+  let l:indentation = '        '
+  call health#report_info("If you're still having problems, try the following commands:\n" .
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: too long line.

File: autoload/health/deoplete.vim [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        \ l:indentation . "$ export NVIM_PYTHON_LOG_FILE=/tmp/log\n" .
+        \ l:indentation . "$ export NVIM_PYTHON_LOG_LEVEL=DEBUG\n" .
+        \ l:indentation . "$ nvim\n" .
+        \ l:indentation . "$ cat /tmp/log_{PID}\n" .
+        \ l:indentation . ' and then create an issue on github'
+        \ )
+endfunction
+
+function! health#deoplete#check() abort
+  call health#report_start('Deoplete.nvim')
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: It should be `deoplete.nvim`.

File: autoload/health/deoplete.vim [hunk 6]
```diff
@@ -1,3 +1,8 @@
+"=============================================================================
+" FILE: deoplete.vim
+" AUTHOR: Shougo Matsushita <Shougo.Matsu at gmail.com>
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: Please add your name.

File: autoload/health/deoplete.vim [hunk 7]
```diff
@@ -13,25 +18,27 @@ function! s:check_required_python_for_deoplete() abort
   else
     call health#report_error('has("python3") was not successful', [
           \ 'Please install the python3 package for neovim.',
-          \ 'A good guide can be found here: https://github.com/tweekmonster/nvim-python-doctor/wiki/Advanced:-Using-pyenv'
+          \ 'A good guide can be found here: ' .
+          \ 'https://github.com/tweekmonster/nvim-python-doctor/wiki/Advanced:-Using-pyenv'
```
Comment (Shougo) [inline autoload/health/deoplete.vim]: Still too long line.
I think you should split the URL like this.
```
'https:foobar'
\ . '/foobar/baz'
```

**Comment by Shougo** [comment]
LGTM.
Thanks.

**Comment by tjdevries** [comment]
@blueyed The main thing I was trying to do was give Shougo an outline that he could add to as people raised common questions for him. Currently it is generic items, but in the future I hope it allows Shougo to not have to answer so many questions.

If you know of any good deoplete specific items, I would be happy to add them!

**Pull request merged**